File: 8kyu/(8 kyu) To square or not/(8 kyu) To square or not.cs
using System;
using System.Collections.Generic;

public class Kata
{
    public static int[] SquareOrSquareRoot(int[] numbers)
    {
        List<int> res = new List<int>();

        foreach (var num in numbers)
        {
            double squaredNumber = Math.Sqrt(num);
            if (unchecked(squaredNumber == (int)squaredNumber))
            {
                res.Add((int)squaredNumber);
            }
            else
            {
                res.Add((int)Math.Pow(num, 2));
            }
        }

        return res.ToArray(); ;
    }
}

File: 8kyu/(8 kyu) To square or not/(8 kyu) To square or not.java
public class Kata {
    public static int[] squareOrSquareRoot(int[] array) {
        int[] res = new int[array.length];

        for (var i = 0; i < array.length; i++) {
            var squaredNumber = Math.sqrt(array[i]);
            if (squaredNumber == (int) squaredNumber) {
                res[i] = (int) squaredNumber;
            } else {
                res[i] = (int) Math.pow(array[i], 2);
            }
        }

        return res;
    }
}
